python/src/main/resources/grpc/python/ipython_server.py

```diff
@@ -146,10 +160,16 @@ def serve(port):
     server.start()
     ipython.start()
     try:
-        while True:
-            time.sleep(_ONE_DAY_IN_SECONDS)
+        while ipython.isKernelAlive():
```

Comment:
We have multiple options when we figure out that the kernel die:
- return an error then kill the process (this pr is doing that)
- restart the kernel ?

The challenge with 1, is that we maybe need to add a new status to the `status` grpc method. like `shutdown`. We can check this status before running like in the `checkIpythonProcessAndRestartIfDead` 

The challenge with 2 is that our execute thread will always be stuck. and we might have no way to kill it.